sys-libs/mesa/mesa-17.1.7.recipe

```diff
@@ -27,119 +28,106 @@ REQUIRES="
 	"
 BUILD_PREREQUIRES="
-	llvm${secondaryArchSuffix}
-	mako_python
-	cmd:bison
-	cmd:find
-	cmd:flex
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
 	cmd:gcc$secondaryArchSuffix
 	cmd:ld$secondaryArchSuffix
-	cmd:scons
-	cmd:sed
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:which
 	"
 
 BUILD()
 {
 	if [ -n "$secondaryArchSuffix" ]; then
 		export HAIKU_SECONDARY_ARCH="$effectiveTargetArchitecture"
 	fi
-	scons $jobArgs build=release
-	#scons $jobArgs build=debug
+
+	autoreconf -fi
+	runConfigure ./configure \
+		--disable-dri \
+		--disable-glx \
+		--disable-glx-tls \
+		--disable-gbm \
+		--disable-gles1 \
+		--disable-gles2 \
+		--disable-shared-glapi \
+		--with-gallium-drivers=swrast \
+		--with-platforms=haiku
+
+	make $jobArgs
 }
 
 INSTALL()
 {
-	mesaBuildDir=build/haiku-$effectiveTargetArchitecture
-
-	mkdir -p $libDir
-	mkdir -p $developLibDir
-
-	# libGL.so makes up the core of our OpenGL kit
-	cp $(find $mesaBuildDir -name 'libGL.so') $libDir
-	symlinkRelative -s $libDir/libGL.so $developLibDir
-	symlinkRelative -s $libDir/libGL.so $libDir/libGL.so.1
+	make install
 
-	# OpenGL ES
-	cp $(find $mesaBuildDir -name 'libEGL.so.1') $libDir
-	symlinkRelative -s $libDir/libEGL.so.1 $libDir/libEGL.so
-	symlinkRelative -s $libDir/libEGL.so.1 $developLibDir/libEGL.so
+	rm $libDir/lib*.la
 
 	# Our rendering add-ons
 	mkdir -p $addOnsDir/opengl
-	cp $(find $mesaBuildDir -name 'libswpipe.so') \
+	mv $(find $libDir -name 'libswpipe.so') \
 		"$addOnsDir/opengl/Software Pipe"
+	rm -rf $libDir/haiku
 
 	# Set some nice version info
 	appVersion=${portVersion//\./ }
 	setversion "$libDir/libGL.so" -app $appVersion -long "Haiku OpenGL kit"
 	setversion "$libDir/libEGL.so.1" -app $appVersion -long "Mesa EGL"
 	setversion "$addOnsDir/opengl/Software Pipe" -app $appVersion -long "Gallium LLVM software pipe renderer"
 
+	prepareInstalledDevelLibs \
+		libGL libEGL
+	fixPkgconfig
+
+	# for compatibility
+	symlinkRelative -s $libDir/libGL.so.0 $libDir/libGL.so
+
 	# OpenGL Kit
 	mkdir -p $includeDir/os/opengl
 	cp ./include/HaikuGL/OpenGLKit.h $includeDir/os/
 	cp ./include/HaikuGL/GLView.h $includeDir/os/opengl/
 	cp ./include/HaikuGL/GLRenderer.h $includeDir/os/opengl/
 
-	# Standard GL headers
-	mkdir -p $includeDir/GL
-	cp ./include/GL/gl.h $includeDir/GL/
-	cp ./include/GL/gl_mangle.h $includeDir/GL/
-	cp ./include/GL/glext.h $includeDir/GL/
-
 	# Create GL symlink in opengl kit
 	symlinkRelative -s $includeDir/GL $includeDir/os/opengl/GL
-
-	# Standard EGL headers
-	mkdir -p $includeDir/EGL
-	cp ./include/EGL/* $includeDir/EGL
-
-	# Create GL symlink in opengl kit
+	# Create EGL symlink in opengl kit
 	symlinkRelative -s $includeDir/EGL $includeDir/os/opengl/EGL
-
-	# KHR headers for EGL
-	mkdir -p $includeDir/KHR
-	cp ./include/KHR/* $includeDir/KHR
-
 	# Create KHR symlink in opengl kit
-	symlinkRelative -s $includeDir/EGL $includeDir/os/opengl/KHR
-
-	mkdir -p $developLibDir/pkgconfig
```

Comment:
I don't remember doing these pkgconfig lines... are we sure fixPkgconfig properly corrects the pkg-config and the autotools install places it down properly?

Reply:
The hpkg is correct. There is already an egl pkgconfig file with autotools, we add the one for libGL.